pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go

```diff
@@ -112,6 +112,26 @@ type BasicAutoScalerSpec struct {
 	// If not set, the default metric will be set to 80% average CPU utilization.
 	// +optional
 	Metrics []v2beta2.MetricSpec `json:"metrics,omitempty"`
+
+	// MetricsTimeWindowSeconds describe the time window seconds for the metrics
+	// to be queried in the Prometheus.
+	// If not set, the default value would be 180.
+	// +optional
+	MetricsTimeWindowSeconds *int32 `json:"metricsTimeWindowSeconds,omitempty"`
+
+	// ScaleOutThreshold describe the consecutive threshold for the auto-scaling,
+	// if the consecutive counts of the scale-out result in auto-scaling reach this number,
+	// the auto-scaling would be performed.
+	// If not set, the default value is 3.
```

Comment:
Nor necessary, users could modify the threshold to be suitable for their Prometheus.

Reply:
For example, Prom scrapes the metrics in 1-minute interval, the data in Prom will be 0,1,2,..., If user configures the window 5m, scaleout threshold 3, auto scaler controller sync interval 30s, then at 04:20 (check metrics of 0-4), 04:50 (check metrics of 0-4), 05:20 (check metrics of 1-5), then it will scale, is this expected?

Reply:
First thing first, we can't configure the Prometheus scrape configure in operator, right? And the factors you listed (time window / threshold / resync duration ) are all configured by users, not calculated by operators. 

So when the auto-scaling performed actually is when the auot-scaling performed expected.